src/net/udpsock.go

```diff
@@ -0,0 +1,249 @@
+// TINYGO: The following is copied and modified from Go 1.19.3 official implementation.
+
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package net
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"tinygo.org/x/drivers/netdev"
+)
+
+// TINYGO: Removed IPv6 stuff
+
+// UDPAddr represents the address of a UDP end point.
+type UDPAddr struct {
+	IP   IP
+	Port int
+}
+
+// Network returns the address's network name, "udp".
+func (a *UDPAddr) Network() string { return "udp" }
+
+func (a *UDPAddr) String() string {
+	if a == nil {
+		return "<nil>"
+	}
+
+	// TINYGO: Work around not having internal/itoa
+
+	ip := []byte(a.IP)
+	if a.Port == 0 {
+		return fmt.Sprintf("%d.%d.%d.%d", ip[0], ip[1], ip[2], ip[3])
+	}
+	return fmt.Sprintf("%d.%d.%d.%d:%d", ip[0], ip[1], ip[2], ip[3], a.Port)
+}
+
+// ResolveUDPAddr returns an address of UDP end point.
+//
+// The network must be a UDP network name.
+//
+// If the host in the address parameter is not a literal IP address or
+// the port is not a literal port number, ResolveUDPAddr resolves the
+// address to an address of UDP end point.
+// Otherwise, it parses the address as a pair of literal IP address
+// and port number.
+// The address parameter can use a host name, but this is not
+// recommended, because it will return at most one of the host name's
+// IP addresses.
+//
+// See func Dial for a description of the network and address
+// parameters.
+func ResolveUDPAddr(network, address string) (*UDPAddr, error) {
+
+	switch network {
+	case "udp", "udp4":
+	default:
+		return nil, fmt.Errorf("Network '%s' not supported", network)
+	}
+
+	// TINYGO: Use netdev resolver
+
+	host, sport, err := SplitHostPort(address)
+	if err != nil {
+		return nil, err
+	}
+
+	port, err := strconv.Atoi(sport)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing port '%s' in address: %s",
+			sport, err)
+	}
+
+	if host == "" {
+		return &UDPAddr{Port: port}, nil
+	}
+
+	ip, err := dev.GetHostByName(host)
+	if err != nil {
+		return nil, fmt.Errorf("Lookup of host name '%s' failed: %s", host, err)
+	}
+
+	return &UDPAddr{IP: ip[:4], Port: port}, nil
+}
+
+// UDPConn is the implementation of the Conn and PacketConn interfaces
+// for UDP network connections.
+type UDPConn struct {
+	fd            netdev.Sockfd
```

Comment:
Similar comment as for netdev.Netdever ... Sockfd should maybe be defined in the TinyGo repo instead of `drivers`.  Also wondering if Sockfd should be `uintptr` instead of `int` (in `os` for instance file descriptors are `uintptr`)

Reply:
int was what socket(2) calls return, so just copied that.  And -1 is a common error return value for socket(2).